cmd/allocator/main.go

```diff
@@ -67,11 +122,30 @@ func main() {
 		ClientCAs:  caCertPool,
 	}
 	srv := &http.Server{
-		Addr:      ":" + port,
+		Addr:      ":" + sslPort,
 		TLSConfig: cfg,
+		// add http OC metrics (opencensus.io/http/server/*)
+		Handler: &ochttp.Handler{
+			Handler: httpsMux,
+		},
 	}
 
-	err = srv.ListenAndServeTLS(tlsDir+"tls.crt", tlsDir+"tls.key")
+	go func() {
+		var err error
+		lock := sync.Mutex{}
+		// force a pod restart if the https server exits.
+		health.AddLivenessCheck("allocator-https", func() error {
```

Comment:
Can we instead having two place for liveliness check and healthz only use healthz and leverage wait group to end the process if ListenAndServeTLS returns?

Reply:
yes good idea, I'm doing it, way simpler :)